### PR TITLE
🐛 [Fix] image size order inconsistencies

### LINKED ISSUE #117: Inconsistent expected image_size [width, height] / [height, width] order in code.
## Describe the bug
So far the expected way, in which one provided the image_size in general.config was in [width, height] order, as all the code in the implementation was expecting it to be provided that way. In the recent changes in commit [main@aba5422](https://github.com/WongKinYiu/YOLO/commit/aba5422d1cfecbb1242e36be806ac82c3b3907b3) with the title "[Add] dynamic image size loader" however, Line 125 in [‎yolo/utils/bounding_box_utils.py](https://github.com/WongKinYiu/YOLO/commit/aba5422d1cfecbb1242e36be806ac82c3b3907b3#diff-5efcc88e1dbfe79bf054c60127be4d5555b0fe3c68fa02d14f262dc4af752f94) got changed from:

`W, H = image_size` to `H, W = image_size` in the generate_anchors function.

In the entire input pipeline however, including the dataloader and data_augmentation, the order is still expected to be in [width, height]. This leads to the bug, when using a non-quadratic image_size, that the input image and the generated anchors (used during training in the loss function and in post-processing during inference) do not match, yielding incorrect results.

## To Reproduce
1. Run lazy.py with the following arguments:
```
task=inference
model=v9-s
use_wandb=False
image_size=[1024,768]
```
2. Check the resulting detections in runs/inference/v9-dev/frame000.png

## Screenshots
Current state:
1024x768_infer_pre-fix

If I change `H, W = image_size` back to `W, H = image_size` at Line125 in ‎yolo/utils/bounding_box_utils.py:
1024x768_infer_post-fix

## Expected behavior
Consistent handling of provided non-quadratic image_size.

## My Recommendation
1. Revert Line 125 in ‎yolo/utils/bounding_box_utils.py back to `W, H = image_size` to solve the issue and keep consistent with the up to this point used [width, height] order definition in config.yaml, as changing everything to [height, width] at this point would require a lot more work.
2. Change the default image_size in config.yaml and in all the tests to some non-quadratic resolution, so that in the future such bugs do not go unnoticed before merging to main (the default is [640, 640] now, with which everything works properly of course)

### PR DESCRIPTION
**Pull request by Adamusen**
Fixes the expected order of image_size to be [width, height] in bounding_box_utils.py/generate_anchors function as in the rest of the repository.

Additionally fixes the same issue in the create_auto_anchor functions of the Vec2Box and Anc2Box classes, where the order does not matter, but for the sake of consistency.

fixes #117

### CODE REVIEW

**Comment by Adamusen** [comment]
Found and fixed two additional bugs:

1, In yolo/tools/solver.py in ValidateModel the PostProcessor received the image size in the wrong order [h, w] instead of [w, h]
2, A spelling mistake, in which in yolo/utils/bounding_box_utils.py inside Anc2Box the update function updated the wrong variable: "self.anchor_grid" instead of "self.anchor_grids" as defined in Anc2Box.__init__().
